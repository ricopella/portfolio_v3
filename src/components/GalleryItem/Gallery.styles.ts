import styled from '@emotion/styled'
import { COLORS } from '../../styles/variables'
import { css } from '@emotion/core'

const ExperienceItemWrapper = styled.div`
  display: grid;
  max-width: 46.875rem;

  // custom width to break on desktop when gets close to Gallery Image
  @media (max-width: 66.1875rem) {
    max-width: 25rem;
  }
`

const GalleryImageWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr;
`

const GalleryImage = css`
  width: 37.5rem;
`

const ExperienceContentWrapper = styled.div`
  display: grid;
  color: ${COLORS.gray.light};
  grid-row-gap: 1rem;
  margin-top: 1rem;
`

const ExperienceTitleRow = styled.div`
  display: grid;
  grid-template-columns: max-content max-content;
  align-items: center;
  grid-column-gap: 0.5rem;

  &:last-child {
    justify-self: end;
  }
`

const ExperienceMyTitleRow = styled.div`
  display: grid;
  grid-template-columns: max-content max-content;
  grid-column-gap: 0.5rem;
`

const ExperienceTitle = styled.div`
  font-size: 1.5rem;
  font-weight: bold;
  color: ${COLORS.gray.copy};
`

const ExperienceMyTitle = styled.div`
  font-size: 1rem;
`

const TagItemsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`

export default {
  ExperienceContentWrapper,
  ExperienceItemWrapper,
  ExperienceMyTitle,
  ExperienceMyTitleRow,
  ExperienceTitle,
  ExperienceTitleRow,
  GalleryImage,
  GalleryImageWrapper,
  TagItemsWrapper,
}
