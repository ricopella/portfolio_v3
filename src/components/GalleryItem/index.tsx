import Image from '../GalleryImage/index'
import React, { FC } from 'react'
import Styled from './Gallery.styles'
import { GalleryProps } from '../../types'

const GalleryItem: FC<GalleryProps> = ({
  alt,
  description,
  filename,
  href,
  span,
  style = {},
  title,
}) => (
  <figure css={Styled.EffectPortfolio}>
    <Image
      filename={filename}
      css={Styled.GalleryImage}
      alt={alt}
      style={{ height: '100%', ...style }}
    />
    <figcaption>
      <h2>
        {title}
        {span ? <span>{span}</span> : null}
      </h2>

      <p>{description}</p>
      <Styled.LinkContainer
        href={href}
        target="_blank"
        rel="noopener noreferrer"
      />
    </figcaption>
  </figure>
)

export default GalleryItem
