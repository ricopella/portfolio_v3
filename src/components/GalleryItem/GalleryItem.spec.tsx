import GalleryItem from './index'
import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

describe('<GalleryItem />', () => {
  const props = {
    alt: 'Portfolio: test',
    description: 'this is a test item',
    href: 'test.com',
    filename: null,
    myTitle: 'Tester',
    tech: ['renderer', 'react'],
    title: 'King Tester',
    year: '2020',
  }

  it('should match GalleryItem snapshot', () => {
    // @ts-ignore
    const internalTree = renderer.create(<GalleryItem {...props} />).toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render children', () => {
    // @ts-ignore
    const { getByText } = render(<GalleryItem {...props} />)

    expect(getByText(/this is a test item/i)).toBeTruthy()
    expect(getByText(/renderer/i)).toBeTruthy()
    expect(getByText(/react/i)).toBeTruthy()
    expect(getByText(/King Tester/i)).toBeTruthy()
    expect(getByText(/2020/i)).toBeTruthy()
  })
})
