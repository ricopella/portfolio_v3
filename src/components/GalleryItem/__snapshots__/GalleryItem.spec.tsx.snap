// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GalleryItem /> should match GalleryItem snapshot 1`] = `
<div
  className="css-1yvep9v"
>
  <div
    className="css-1xvo91r"
  >
    <div />
  </div>
  <div
    className="css-1hg6sd7"
  >
    <div
      className="css-d8eafh"
    >
      <div
        className="css-1ijvccx"
      >
        King Tester
      </div>
      <a
        className="css-187n1lu"
        href="test.com"
        rel="noopener noreferrer"
        target="_blank"
        title="test.com"
      >
        <div
          className="css-3nfvs0"
        />
      </a>
    </div>
    <div
      className="css-15fkjsq"
    >
      <div
        className="css-11aljab"
      >
        Tester
      </div>
      <div
        className="css-11aljab"
      >
        |
         
        2020
      </div>
    </div>
    <p>
      this is a test item
    </p>
    <div
      className="css-1p6kesl"
    >
      <div
        className="css-3m0mjb"
      >
        <div
          className="css-cjmkt3"
        >
          renderer
        </div>
      </div>
      <div
        className="css-3m0mjb"
      >
        <div
          className="css-cjmkt3"
        >
          react
        </div>
      </div>
    </div>
  </div>
</div>
`;
